package.json

```diff
@@ -30,6 +30,7 @@
     "tap": "^14.3.1"
   },
   "dependencies": {
-    "fastify-plugin": "^1.6.0"
+    "fastify-plugin": "^1.6.0",
+    "js-xxhash": "^1.0.1"
```

Comment:
I never used them, but why not optionalDependencies ?
http://npm.github.io/using-pkgs-docs/package-json/types/optionaldependencies.html

Reply:
I was thinking about `peerDependencies` but this might be more appropriate.

The only thing I am a bit concerned is that it will make the actual plugin code harder and we will also have to find a way to manage potentially compatible/incompatible versions of the module in the code itself...

I am more for keeping this as a regular dependency for now. Of course, if this plugin keeps growing to support many many more algorithms (and dependencies), maybe it will make more sense to take such an approach.